src/auto_edit.rs

```diff
@@ -0,0 +1,279 @@
+            ui.add(egui::DragValue::new(&mut value.y).prefix("y: ").speed(0.1));
+            ui.add(egui::DragValue::new(&mut value.z).prefix("z: ").speed(0.1));
+            ui.add(egui::DragValue::new(&mut value.w).prefix("w: ").speed(0.1));
+        });
+    }
+}
+
+impl YoleckAutoEdit for Quat {
+    fn auto_edit(value: &mut Self, ui: &mut egui::Ui) {
+        let (mut yaw, mut pitch, mut roll) = value.to_euler(EulerRot::YXZ);
+        yaw = yaw.to_degrees();
+        pitch = pitch.to_degrees();
+        roll = roll.to_degrees();
+        
+        ui.horizontal(|ui| {
+            let mut changed = false;
+            changed |= ui.add(egui::DragValue::new(&mut yaw).prefix("yaw: ").speed(1.0).suffix("°")).changed();
+            changed |= ui.add(egui::DragValue::new(&mut pitch).prefix("pitch: ").speed(1.0).suffix("°")).changed();
+            changed |= ui.add(egui::DragValue::new(&mut roll).prefix("roll: ").speed(1.0).suffix("°")).changed();
+            
+            if changed {
+                *value = Quat::from_euler(
+                    EulerRot::YXZ,
+                    yaw.to_radians(),
+                    pitch.to_radians(),
+                    roll.to_radians(),
+                );
+            }
+        });
+    }
+}
+
+impl YoleckAutoEdit for Color {
+    fn auto_edit(value: &mut Self, ui: &mut egui::Ui) {
+        let srgba = value.to_srgba();
+        let mut color_arr = [srgba.red, srgba.green, srgba.blue, srgba.alpha];
+        if ui.color_edit_button_rgba_unmultiplied(&mut color_arr).changed() {
+            *value = Color::srgba(color_arr[0], color_arr[1], color_arr[2], color_arr[3]);
+        }
+    }
+}
+
+impl<T: YoleckAutoEdit + Default> YoleckAutoEdit for Option<T> {
+    fn auto_edit(value: &mut Self, ui: &mut egui::Ui) {
+        ui.horizontal(|ui| {
+            let mut has_value = value.is_some();
+            if ui.checkbox(&mut has_value, "").changed() {
+                if has_value {
+                    *value = Some(T::default());
+                } else {
+                    *value = None;
+                }
+            }
+            if let Some(ref mut inner) = value {
+                T::auto_edit(inner, ui);
+            }
+        });
+    }
+}
+
+impl<T: YoleckAutoEdit + Default> YoleckAutoEdit for Vec<T> {
+    fn auto_edit(value: &mut Self, ui: &mut egui::Ui) {
+        let mut to_remove = None;
+        for (idx, item) in value.iter_mut().enumerate() {
+            ui.horizontal(|ui| {
+                ui.label(format!("[{}]", idx));
+                T::auto_edit(item, ui);
+                if ui.small_button("−").clicked() {
+                    to_remove = Some(idx);
+                }
+            });
+        }
+        if let Some(idx) = to_remove {
+            value.remove(idx);
+        }
+        if ui.small_button("+").clicked() {
+            value.push(T::default());
+        }
+    }
+}
+
+impl<const N: usize, T: YoleckAutoEdit + Default> YoleckAutoEdit for [T; N] {
```

Comment:
I wonder - can you implement it instead for `[T]`? It would allow for non-constant arrays that the user is not allowed to add or remove items from them.

This would mean, of course, that `YoleckAutoEdit` would need to be `?Sized`.